pkg/app/piped/executor/ecs/ecs.go

```diff
@@ -432,17 +434,32 @@ func routing(ctx context.Context, in *executor.Input, platformProviderName strin
 		},
 	}
 
-	metadata := map[string]string{
+	metadataPercentage := map[string]string{
 		trafficRoutePrimaryMetadataKey: strconv.FormatInt(int64(primary), 10),
 		trafficRouteCanaryMetadataKey:  strconv.FormatInt(int64(canary), 10),
 	}
-	if err := in.MetadataStore.Stage(in.Stage.Id).PutMulti(ctx, metadata); err != nil {
+	if err := in.MetadataStore.Stage(in.Stage.Id).PutMulti(ctx, metadataPercentage); err != nil {
 		in.Logger.Error("Failed to store traffic routing config to metadata store", zap.Error(err))
 	}
 
-	currListenerArns, err := client.GetListenerArns(ctx, primaryTargetGroup)
-	if err != nil {
-		in.LogPersister.Errorf("Failed to get current active listeners: %v", err)
+	var currListenerArns []string
+	value, ok := in.MetadataStore.Shared().Get(currentListenersKey)
+	if ok {
+		currListenerArns = strings.Split(value, ",")
+	} else {
+		currListenerArns, err = client.GetListenerArns(ctx, primaryTargetGroup)
+		if err != nil {
+			in.LogPersister.Errorf("Failed to get current active listeners: %v", err)
+			return false
+		}
+	}
+
+	// Store created listeners to use later.
+	addedListeners := make([]string, 0, len(currListenerArns))
+	addedListeners = append(addedListeners, currListenerArns...)
+	metadata := strings.Join(addedListeners, ",")
+	if err := in.MetadataStore.Shared().Put(ctx, currentListenersKey, metadata); err != nil {
+		in.LogPersister.Errorf("Unable to store created listeners to metadata store: %v", err)
 		return false
 	}
```

Comment:
Thank you! I fixed it. [use currListenerArns instead of addedListeners](https://github.com/pipe-cd/pipecd/pull/4599/commits/87433dbe215e26d4dd73e3672fe516a3cd46dc45)